jni/src/nmslib_wrapper.cpp

```diff
@@ -317,25 +323,16 @@ jobjectArray knn_jni::nmslib_wrapper::QueryIndex(knn_jni::JNIUtilInterface *jniU
   }
 
   int queryEfSearch = knn_jni::commons::getIntegerMethodParameter(env, jniUtil, methodParams, EF_SEARCH, -1);
-  similarity::KNNQuery<float>
-      *query; // TODO: Replace with smart pointers https://github.com/opensearch-project/k-NN/issues/1785
+  std::unique_ptr<similarity::KNNQuery<float>> query;
   std::unique_ptr<similarity::KNNQueue<float>> neighbors;
-  try {
-    if (queryEfSearch == -1) {
-      query = new similarity::KNNQuery<float>(*(indexWrapper->space), queryObject.get(), kJ);
-    } else {
-      query = new similarity::HNSWQuery<float>(*(indexWrapper->space), queryObject.get(), kJ, queryEfSearch);
-    }
-
-    indexWrapper->index->Search(query);
-    neighbors.reset(query->Result()->Clone());
-  } catch (...) {
-    if (query != nullptr) {
-      delete query;
-    }
-    throw;
+  if (queryEfSearch == -1) {
+    query.reset(new similarity::KNNQuery<float>(*(indexWrapper->space), queryObject.get(), kJ));
+  } else {
+    query.reset(new similarity::HNSWQuery<float>(*(indexWrapper->space), queryObject.get(), kJ, queryEfSearch));
   }
-  delete query;
+
+  indexWrapper->index->Search(query.get());
```

Comment:
Do we need to touch search code in this change?

Reply:
No!
I thought it would be a simple fix to have it in this PR.
If you're concern, then I can revert it back.

Reply:
Sure - Id say dont revert now. In future, especially for C++ layer changes though, I would try to stick to the scope.

Reply:
sure, for the next time, I will restrict PR to be within its own scope!